--- ggobi-2.1.9/plugins/GraphLayout/graphviz.c.orig	2012-02-20 00:29:02.554331731 -0200
+++ ggobi-2.1.9/plugins/GraphLayout/graphviz.c	2012-02-20 00:29:32.031441888 -0200
@@ -267,8 +267,8 @@ void dot_neato_layout_cb (GtkWidget *but
       m = visible[i];
       name = (gchar *) g_array_index (d->rowlab, gchar *, m);
       node = agfindnode (graph, name);
-      pos[i][0] = ND_coord_i(node).x;
-      pos[i][1] = ND_coord_i(node).y;
+      pos[i][0] = (node)->u.coord.x;
+      pos[i][1] = (node)->u.coord.y;
     }
 
   } else if (layout_type == TWOPI_LAYOUT) {
