--- ggobi-2.1.9/src/ggobi.c.orig	2012-02-19 23:44:58.151468338 -0200
+++ ggobi-2.1.9/src/ggobi.c	2012-02-19 23:45:38.678617049 -0200
@@ -698,10 +698,10 @@ ValidateGGobiRef (ggobid * gg, gboolean
   }
 
   if (fatal) {
-    g_error (error_msg);
+    g_error ("%s", error_msg);
   }
   else
-    g_critical (error_msg);
+    g_critical ("%s", error_msg);
 
   return (NULL);
 }
@@ -718,9 +718,9 @@ ValidateDatadRef (GGobiData * d, ggobid
   }
 
   if (fatal)
-    g_error (error_msg);
+    g_error ("%s", error_msg);
   else
-    g_critical (error_msg);
+    g_critical ("%s", error_msg);
 
   return (NULL);
 }
@@ -739,9 +739,9 @@ ValidateDisplayRef (displayd * d, ggobid
   }
 
   if (fatal)
-    g_error (error_msg);
+    g_error ("%s", error_msg);
   else
-    g_critical (error_msg);
+    g_critical ("%s", error_msg);
 
   return (NULL);
 }
--- ggobi-2.1.9/src/utils_ui.c.orig	2012-02-19 23:48:41.312287053 -0200
+++ ggobi-2.1.9/src/utils_ui.c	2012-02-19 23:48:57.040344742 -0200
@@ -189,7 +189,7 @@ quick_message (const gchar * const messa
 
   dialog =
     gtk_message_dialog_new (NULL, 0, GTK_MESSAGE_INFO, GTK_BUTTONS_OK,
-                            message);
+                            "%s", message);
 
   if (modal)
     gtk_window_set_modal (GTK_WINDOW (dialog), true);
--- ggobi-2.1.9/src/write_xml.c.orig	2012-02-19 23:50:51.856765819 -0200
+++ ggobi-2.1.9/src/write_xml.c	2012-02-19 23:51:13.487845138 -0200
@@ -37,14 +37,14 @@ static void
 write_xml_string(FILE *f, gchar *str)
 {
   gchar *fmtstr = g_markup_printf_escaped("%s", str);
-  fprintf(f, fmtstr);
+  fprintf(f, "%s", fmtstr);
   g_free(fmtstr);
 }
 static void
 write_xml_string_fmt(FILE *f, gchar *fmt, gchar *str)
 {
   gchar *fmtstr = g_markup_printf_escaped(fmt, str);
-  fprintf(f, fmtstr);
+  fprintf(f, "%s", fmtstr);
   g_free(fmtstr);
 }
 
